common/tempstorage/storage.go

```diff
@@ -0,0 +1,44 @@
+package tempstorage
+
+import "io"
+
+type storage interface {
```

Comment:
ok to export ?  Can also add
var _ tempstorage.Storage = DiskStorage{} inside the diskstore and memstore packages to verify that those other packages are implementing interfaces correctly, i.e. a compile time check.

Reply:
I tried to make Open, TempDir etc. methods of a package, not of a storage object (like os.Open, ioutil.TempFile etc), so is this a reason to make it exported?